### DISCUSSION TITLE
Customizing meow-next-word

### DISCUSSION THREAD

**Discussion by joshcho** [meow-edit/meow#184] [Q&A] [answered]
Currently, when calling `meow-next-word` when my cursor is at `apply-to-arg| modes`, the region selected becomes `apply-to-arg| modes|`. Is there a way to make it such that `meow-next-word` just selects `apply-to-arg |modes|`? I have looked at the code for `meow-next-word`, but would love some pointers on how to modify it for my own tastes. Would be greatly appreciated.

EDIT: I have something like:
`(lambda () (interactive) (forward-word) (backward-word) (meow-next-word 1)))`, which works but seems inefficient.

**Comment by DogLooksGood** [comment, marked as answer]
Yes, the only way is having your own implementation, there's no such option.

```
(defun meow-next-word (n)
  (interactive "p")
  (unless (equal 'word (cdr (meow--selection-type)))
    (meow--cancel-selection))
  (let* ((expand (equal '(expand . word) (meow--selection-type)))
         (_ (when expand (meow--direction-forward)))
         (type (if expand '(expand . word) '(select . word)))
         (p (save-mark-and-excursion
              (when (forward-word n)
                (point))))
	 (m (save-mark-and-excursion
	      (goto-char p)
	      (backward-word)
	      (point))))
    (when p
      (thread-first
        (meow--make-selection type (meow--fix-word-selection-mark p m) p expand)
        (meow--select))
      (meow--maybe-highlight-num-positions '(meow--backward-word-1 . meow--forward-word-1)))))
```
A modified version based on current implementation, `m` in let binding is the mark which is the position where your selection start. 

---

If you don't mind press one more key, use `meow-mark-word` after `meow-next-word`. It's designed to be used in this case.
